src/helpers/AvroToMetaModel.ts

```diff
@@ -0,0 +1,394 @@
+    avroSchemaModel.type.length >= 8 &&
+    avroSchemaModel.type !== null;
+  const containsAllTypes =
+    Array.isArray(avroSchemaModel.type) && avroSchemaModel.type.length === 10;
+  return containsAllTypesButNotNull || containsAllTypes;
+}
+
+export function AvroToMetaModel(
+  avroSchemaModel: AvroSchema,
+  alreadySeenModels: Map<AvroSchema, MetaModel> = new Map()
+): MetaModel {
+  const hasModel = alreadySeenModels.has(avroSchemaModel);
+  if (hasModel) {
+    return alreadySeenModels.get(avroSchemaModel) as MetaModel;
+  }
+
+  const modelName = avroSchemaModel.name || 'undefined';
+
+  if (shouldBeAnyType(avroSchemaModel)) {
+    return new AnyModel(
+      modelName,
+      avroSchemaModel.originalInput,
+      getMetaModelOptions(avroSchemaModel)
+    );
+  }
+  const objectModel = toObjectModel(
+    avroSchemaModel,
+    modelName,
+    alreadySeenModels
+  );
+  if (objectModel !== undefined) {
+    return objectModel;
+  }
+  const arrayModel = toArrayModel(
+    avroSchemaModel,
+    modelName,
+    alreadySeenModels
+  );
+  if (arrayModel !== undefined) {
+    return arrayModel;
+  }
+  const booleanModel = toBooleanModel(avroSchemaModel, modelName);
+  if (booleanModel !== undefined) {
+    return booleanModel;
+  }
+  const stringModel = toStringModel(avroSchemaModel, modelName);
+  if (stringModel !== undefined) {
+    return stringModel;
+  }
+  const integerModel = toIntegerModel(avroSchemaModel, modelName);
+  if (integerModel !== undefined) {
+    return integerModel;
+  }
+  const floatModel = toFloatModel(avroSchemaModel, modelName);
+  if (floatModel !== undefined) {
+    return floatModel;
+  }
+  const enumModel = toEnumModel(avroSchemaModel, modelName);
+  if (enumModel !== undefined) {
+    return enumModel;
+  }
+  const unionModel = toUnionModel(
+    avroSchemaModel,
+    modelName,
+    alreadySeenModels
+  );
+  if (unionModel !== undefined) {
+    return unionModel;
+  }
+
+  Logger.warn('Failed to convert to MetaModel, defaulting to AnyModel.');
+  return new AnyModel(
+    modelName,
+    avroSchemaModel.originalInput,
+    getMetaModelOptions(avroSchemaModel)
+  );
+}
+
+export function toBooleanModel(
+  avroSchemaModel: AvroSchema,
+  name: string
+): BooleanModel | undefined {
+  if (avroSchemaModel.type !== 'boolean') {
+    return undefined;
+  }
+  return new BooleanModel(
+    name,
+    avroSchemaModel.originalInput,
+    getMetaModelOptions(avroSchemaModel)
+  );
+}
+export function toIntegerModel(
+  avroSchemaModel: AvroSchema,
+  name: string
+): IntegerModel | undefined {
+  if (avroSchemaModel.type !== 'int' && avroSchemaModel.type !== 'long') {
+    return undefined;
+  }
+  return new IntegerModel(
+    name,
+    avroSchemaModel.originalInput,
+    getMetaModelOptions(avroSchemaModel)
+  );
+}
+export function toFloatModel(
+  avroSchemaModel: AvroSchema,
+  name: string
+): FloatModel | undefined {
+  if (avroSchemaModel.type !== 'float' && avroSchemaModel.type !== 'double') {
```

Comment:
This one is a bit hard, because we have two inputs that have a `type` keyword. JSON Schema and Avro, each with overlapping types such as `string` and `boolean`.

So the answer to the question lies within "what is the difference between Avro and JSON Schema and how can you easily detect it" 😄 

I don't have the perfect answer here unfortunately, other then look at the keywords that are different between them and match with those 🙂

Reply:
> @jonaslagoni How about introducing a check explicitly in the shouldProcess() function using an additional schema property with a value of Avro to differentiate an Avro Schema before processing the input?

Can you give an example of what you mean?

Reply:
Example: 

{
  "name": "Person",
  "type": "int",
  "schema": "avro"
}

Here, we can add a check for the `schema` property against the value `avro`. This would be to verify that only avro schema is passed to the Avro processor.

Reply:
That would never happen no, because `schema` is not part of the Avro standard and never will be probably 🙂 

It's okay if the solution is not bullet proof, as long as we document the edge cases.